sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/identity/credential/provider/DefaultTokenCredentialProvider.java

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.service.implementation.identity.credential.provider;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.identity.ClientCertificateCredential;
+import com.azure.identity.ClientCertificateCredentialBuilder;
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.azure.identity.DefaultAzureCredential;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.identity.ManagedIdentityCredential;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.azure.identity.UsernamePasswordCredential;
+import com.azure.identity.UsernamePasswordCredentialBuilder;
+import com.azure.spring.cloud.service.implementation.identity.credential.TokenCredentialProvider;
+import com.azure.spring.cloud.service.implementation.identity.credential.TokenCredentialProviderOptions;
+import com.azure.spring.cloud.service.implementation.identity.StaticAccessTokenCache;
+import com.azure.spring.cloud.service.implementation.identity.credential.CacheableTokenCredential;
+import com.azure.spring.cloud.service.implementation.identity.credential.adapter.CacheableClientCertificateCredential;
+import com.azure.spring.cloud.service.implementation.identity.credential.adapter.CacheableClientSecretCredential;
+import com.azure.spring.cloud.service.implementation.identity.credential.adapter.CacheableDefaultAzureCredential;
+import com.azure.spring.cloud.service.implementation.identity.credential.adapter.CacheableManageIdentityCredential;
+import com.azure.spring.cloud.service.implementation.identity.credential.adapter.CacheableUsernamePasswordCredential;
+import org.springframework.util.StringUtils;
+
+/**
+ * Default tokenCredentialProvider implementation that provides tokenCredential instance.
+ */
+public class DefaultTokenCredentialProvider implements TokenCredentialProvider {
+
+    private final TokenCredentialProviderOptions options;
+    private final StaticAccessTokenCache cache = new StaticAccessTokenCache();
```

Comment:
it is `StaticAccessTokenCache`, so, probably making it singleton

Reply:
Ok